services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/accessSurrogates/AccessSurrogateRequestProcessor.java

```diff
@@ -46,20 +48,24 @@ public void process(Exchange exchange) throws Exception {
 
         var action = request.getAction();
         var repositoryObject = repositoryObjectLoader.getRepositoryObject(pid);
+        var filePath = request.getFilePath();
         if (repositoryObject instanceof FileObject) {
-
             Path surrogatePath = derivativeService.getDerivativePath(pid, DatastreamType.ACCESS_SURROGATE);
             if (Objects.equals(SET, action)) {
-                var filePath = request.getFilePath();
                 Files.copy(filePath, surrogatePath, StandardCopyOption.REPLACE_EXISTING);
             } else if (Objects.equals(DELETE, action)) {
-                deleteFile(surrogatePath);
+                Files.deleteIfExists(surrogatePath);
             }
 
             Document msg = makeEnhancementOperationBody(agent.getUsername(), pid, true);
             messageSender.sendMessage(msg);
+        } else {
+            log.warn("Cannot process access surrogate update for {}, non FileObjects do not have access surrogates",
+                    pid.getId());
         }
 
+        // clean up file from request
+        Files.deleteIfExists(filePath);
```

Comment:
I think this might throw an error during a DELETE action since the filePath won't be set? So might need to wrap it in a null check.

Reply:
ahh good catch!